tools/ci_connector_ops/ci_connector_ops/pipelines/tests/java_connectors.py

```diff
@@ -0,0 +1,111 @@
+        "airbyte-db",
+        "airbyte-integrations/bases",
+        "airbyte-json-validation",
+        "airbyte-protocol",
+        "airbyte-test-utils",
+        "buildSrc",
+        "tools/bin/build_image.sh",
+        "tools/lib/lib.sh",
+    ]
+
+    SOURCE_BUILD_INCLUDE = [
+        "airbyte-integrations/connectors/source-jdbc",
+        "airbyte-integrations/connectors/source-relational-db",
+    ]
+
+    DESTINATION_BUILD_INCLUDE = ["pyproject.toml"]  # For normalization airbyte-python
+
+    @property
+    def build_include(self) -> List[str]:
+        if self.context.connector.connector_type == "source":
+            return self.JAVA_BUILD_INCLUDE + self.SOURCE_BUILD_INCLUDE
+        else:
+            return self.JAVA_BUILD_INCLUDE + self.DESTINATION_BUILD_INCLUDE
+
+    @property
+    def title(self) -> str:
+        return f"Gradle {self.task_name} task"
+
+    async def get_patched_connector_dir(self) -> Directory:
+        """Removes the airbyte-connector-acceptance-test plugin import from build.gradle to not run CAT with Gradle.
+
+        Returns:
+            Directory: The patched connector directory
+        """
+        gradle_file_content = await self.context.get_connector_dir(include=["build.gradle"]).file("build.gradle").contents()
+        patched_file_content = ""
+        for line in gradle_file_content.split("\n"):
+            if "id 'airbyte-connector-acceptance-test'" not in line:
+                patched_file_content += line + "\n"
+        return self.context.get_connector_dir(exclude=["build", "secrets"]).with_new_file("build.gradle", patched_file_content)
+
+    def get_gradle_command(self, extra_options: Tuple[str] = ("--no-daemon", "--scan")) -> List:
+        return (
+            ["./gradlew"]
+            + list(extra_options)
+            + [f":airbyte-integrations:connectors:{self.context.connector.technical_name}:{self.task_name}"]
+        )
+
+    async def _run(self) -> StepResult:
+        self.context.dagger_client = self.get_dagger_pipeline(self.context.dagger_client)
+        connector_java_build_cache = self.context.dagger_client.cache_volume("connector_java_build_cache")
+        connector_under_test = (
+            environments.with_gradle(self.context, self.build_include, docker_cache_volume_name="docker-lib-gradle")
+            .with_mounted_cache(f"{self.context.connector.code_directory}/build", connector_java_build_cache)
+            .with_mounted_directory(str(self.context.connector.code_directory), await self.get_patched_connector_dir())
+            # Disable the Ryuk container because it needs privileged docker access that does not work:
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true")
+            .with_directory(f"{self.context.connector.code_directory}/secrets", self.context.secrets_dir)
+            .with_exec(self.get_gradle_command())
+        )
+
+        return await self.get_step_result(connector_under_test)
+
+
+class Test(GradleTask):
```

Comment:
I dont imagine we would be doing a lot of overriding. I'd suggest removing these sub classes in favor of explictly passing in the gradle command

```py
GradleTask(context).run("test")
```